### ISSUE TITLE
CF-1.10, CF-1.11 availability?

### ISSUE THREAD

**Issue by jcermauwedu** [ioos/compliance-checker#1110]
@ocefpaf The Open Gliders 1.0 format has been approved by the community this past summer.  This format requires compliance checking at CF-1.10.   It would be great to see CF-1.10+ available from the main checker if there are programming cycles available. https://github.com/uw-farlab/cc-plugin-og/pull/3#issuecomment-2393682487.

REF: https://oceangliderscommunity.github.io/OG-format-user-manual/OG_Format.html

Although there is an online checker for the OG-1.0 format for testing, http://conventions.castelao.net/OG/1.0/validate, it appears to be checking against CF-1.11.  There was a slight indication that the rust source code for this was to be available in the future.

**Comment by jcermauwedu** [comment]
Yes, IIRC, the base CF-1.10+ items are there I believe.  I think it was a matter of including the unstructured grid components, https://github.com/ioos/cc-plugin-ugrid, to make it a complete upgrade.  Or keep the ugrid items in its own plugin.

**Comment by ocefpaf** [comment]
> Or keep the ugrid items in its own plugin.

If it is CF we should probably merge those checks into main compliance-checker, not an easy feat though 😬

**Comment by callumrollo** [comment]
Hi compliance checker team! Has there been any movement on CF-1.10 support? I'm allocating some time to improving the [cc-plugin-og](https://github.com/uw-farlab/cc-plugin-og) that @jcermauwedu mentioned. compliance checker support for CF-1.10 is essential for this.

Thanks!

**Comment by benjwadams** [comment]
There are a number of sections of CF conformance that need to be fixed or added to in some capacity to support CF.  See the various open "Improve CF Conformance" issues posted in GitHub.

I have a meeting with @mwengren coming up soon and I'll discuss with him.

**Comment by benjwadams** [comment]
To have full support for CF 1.10 etc we'd have to implement some sections of CF that I haven't seen used much in practice, but are non-trivial to implement, such as https://github.com/ioos/compliance-checker/issues/957

People are welcome to submit contributions, but in the short term, it's likely that we'll release CF 1.10 and 1.11 checkers without some of these features but denote where the features are missing in the README and similar documentation.

**Comment by ocefpaf** [comment]
@jcermauwedu do you have the links for the Google docs from the code sprint? We are starting to list the missing features to deicide on that prioritize for CF>=1.10.

**Comment by jcermauwedu** [comment]
@ocefpaf Here are the available notes:
- https://ioos.github.io/ioos-code-sprint/2024/topics/02-compliance-checker-topics.html
- https://github.com/ioos/ioos-code-sprint/issues/40

**Comment by ocefpaf** [comment]
BTW, we created a milestone on GH for it and it is evolving quite fast, giving that a lot was already implemented, and some parts are not worth tackling as no checker does, like the compression rules, etc.

**Comment by benjwadams** [comment]
Honestly, 1.9 to 1.10 doesn't appear to have actually needed code changes -- the changes appear to have been various clarifications within the CF 1.10 docs when I checked for #1164.   The small issue I did find for #1158 is just a clarification of an optional coordinate for `featureType` and we don't currently check `coordinates` for `featureType` associated variables, per #1163.

**Comment by benjwadams** [comment]
Going to close this in favor of the existing milestones. #1158 was just merged and a new version released so 1.10 should be supported, even though it is the same as CF 1.9 in terms of code implementation. If you have additional issues to bring up pertinent to the versions mentioned in this issue, please create focused, separate issues which we can integrate into milestones.
